api/app/signals/apps/api/v1/serializers.py

```
@@ -47,38 +47,88 @@
-    category = CategoryHyperlinkedRelatedField(write_only=True, required=True)
-    state_display = serializers.CharField(source='get_state_display', read_only=True)
+    def to_representation(self, data):
+        representation = super(StateStatusMessageTemplateLisSerializer, self).to_representation(
+            data=data
+        )
+        return self._transform_representation(representation)
+
+    def to_internal_value(self, data):
+        status_template_messages = []
+        for item in data:
+            status_template_message = {
+                'state': item['state'],
+                'category': self.context['category'],
+            }
+
+            if len(item['templates']):
+                order = 0
+                for template in item['templates']:
+                    status_template_message_copy = copy.copy(status_template_message)
+                    status_template_message_copy.update({
+                        'title': template['title'],
+                        'text': template['text'],
+                        'order': order,
+                    })
+                    status_template_messages.append(status_template_message_copy)
+                    order += 1
+            else:
+                status_template_messages.append(status_template_message)
+        return status_template_messages
+
+    def save(self, **kwargs):
+        # Delete all status templates for each state we are adding, because we do a overwrite of
+        # the complete set
+        valid_states = [valid_data['state'] for valid_data in self.validated_data]
+        StatusMessageTemplate.objects.filter(state__in=valid_states,
+                                             category=self.context['category']).delete()
+
+        # Actual creation of the status message templates
+        for valid_data in self.validated_data:
+            # If there is no title and no text we cannot add the template
```

These empty status messages are silently ignored. Probably ok, but should check with frontend people whether they need it. (No blocker.)